test/known_problems/should_pass/guard_negative_refinement_should_pass.erl

```diff
@@ -0,0 +1,15 @@
+-module(guard_negative_refinement_should_pass).
```

Comment:
Negative? I believe `not_equal` is better when talking about the `/=` operator.

Btw, do we have tests for guards with `=/=`? And what about `==` and `=:=`?

Reply:
Hmm, the *negative* part was meant like *type negation* (e.g. `(integer() | nil) \ nil`) but this is already contained in the *refinement*… So maybe it should be only `guard_refinement.erl`? There's one file that already tests guard refinements (including `=/=`, `==`, and `=:=`), `test/should_pass/refine_comparison.erl`, so maybe I should merge the files and put it there? What do you think?

Reply:
Yeah... merge the two, or rename the other one, or both, or what you find appropriate. :) Now you have the full picture.

Reply:
Okay. :) I merged it into the `test/should_pass/refine_comparison.erl` and created `test/known_problems/should_pass/refine_comparison_should_pass.erl` with the failing test. I've also squashed the fixups and force pushed.